### PR TITLE
List of all miners available in the system

### PR DESCRIPTION
**Pull request by rajakvk**
Implementation of the [issue](https://github.com/Real-Dev-Squad/website-backend/issues/360)

### CODE REVIEW

File: models/miners.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const firestore = require('../utils/firestore')
+const minersModel = firestore.collection('miners')
+
+const fetchMiners = async (size = 100, page = 0) => {
+  try {
+    const snapshot = await minersModel
+      .limit(parseInt(size))
```
Comment (swarajpure) [inline models/miners.js:7]: Please add JSDoc

File: routes/miners.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+const express = require('express')
+const router = express.Router()
+const miners = require('../controllers/miners.js')
+
+router.get('/list', miners.getMiners)
```
Comment (swarajpure) [inline routes/miners.js:5]: Please add swagger docs

**Review by swarajpure — CHANGES_REQUESTED** [review]
JSDoc and Swagger docs are missing, please add them 🙂

File: models/miners.js (hunk 1 above)
Comment (rajakvk) [reply models/miners.js]: Added JSDoc

File: routes/miners.js (hunk 2 above)
Comment (rajakvk) [reply routes/miners.js:35]: Added swagger docs

File: controllers/miners.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+const getMiners = async (req, res) => {
+  try {
+    const allMiners = await minersModel.fetchMiners()
+    return res.json({
+      message: 'Miners returned successfully!',
+      miners: allMiners
+    })
+  } catch (error) {
+    logger.error(`Error while fetching all miners: ${error}`)
+    return res.boom.serverUnavailable('Something went wrong please contact admin')
```
Comment (ankurnarkhede) [inline controllers/miners.js:17]: return 500.

File: routes/miners.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+ *                type: object
+ *                properties:
+ *                  message:
+ *                    type: string
+ *                    example: Miners returned successfully!
+ *                  miners:
+ *                    type: array
+ *                    items:
+ *                      $ref: '#/components/schemas/miners'
+ *       503:
```
Comment (ankurnarkhede) [inline routes/miners.js:28]: Return 500.

**Review by ankurnarkhede — CHANGES_REQUESTED** [review]
- Need to change 503 to 500 in error response.
- Unit and integration tests required

**Comment by ankurnarkhede** [comment]
@rajakvk , can you also link the corresponding issue here?

File: controllers/miners.js (hunk 3 above)
Comment (rajakvk) [reply controllers/miners.js:17]: changed to 
return res.boom.badImplementation('An internal server error occured.')

File: routes/miners.js (hunk 4 above)
Comment (rajakvk) [reply routes/miners.js:28]: updated accordingly

**Comment by rajakvk** [comment]
This PR implements [issue](https://github.com/Real-Dev-Squad/website-backend/issues/360).

**Pull request closed without merging**